apps/studio/components/ui/EnvironmentMetadata.tsx

```diff
@@ -0,0 +1,76 @@
+import { useEffect } from 'react'
+
+function EnvironmentMetadata() {
+  const environment = process.env.NEXT_PUBLIC_ENVIRONMENT
+  const commitSha = process.env.NEXT_PUBLIC_VERCEL_GIT_COMMIT_SHA
+  const commitMessage = process.env.NEXT_PUBLIC_VERCEL_GIT_COMMIT_MESSAGE
+  const pullRequestId = process.env.NEXT_PUBLIC_VERCEL_GIT_PULL_REQUEST_ID
+  const branch = process.env.NEXT_PUBLIC_VERCEL_GIT_COMMIT_REF
+  const repoOwner = process.env.NEXT_PUBLIC_VERCEL_GIT_REPO_OWNER
+  const repoSlug = process.env.NEXT_PUBLIC_VERCEL_GIT_REPO_SLUG
+
+  const isPlatform = process.env.NEXT_PUBLIC_IS_PLATFORM
+
+  const repoUrl = `https://github.com/${repoOwner}/${repoSlug}`
+
+  useEffect(() => {
+    if ((environment === 'staging' || environment === 'local') && isPlatform) {
+      console.log(
+        `\n⚡️ ⚡️ ⚡️\n\n%cSupabase Dashboard\n\n%cEnvironment: %c${environment || 'undefined'}\n%cCommit SHA: %c${commitSha ? `${repoUrl}/commit/${commitSha}` : 'undefined'}\n%cCommit Message: %c${commitMessage || 'undefined'}\n%cPull Request ID: %c${pullRequestId ? `${repoUrl}/pull/${pullRequestId}` : 'undefined'}\n%cBranch: %c${branch ? `${repoUrl}/tree/${branch}` : 'undefined'}\n%cRepository: %c${repoOwner || 'undefined'}/${repoSlug || 'undefined'}\n\n`,
+        'color: green; font-size: 16px; font-weight: bold;',
+        'color: inherit;',
+        'color: green;',
+        'color: inherit;',
+        'color: green;',
+        'color: inherit;',
+        'color: green;',
+        'color: inherit;',
+        'color: green;',
+        'color: inherit;',
+        'color: green;',
+        'color: inherit;',
+        'color: green;'
+      )
+    }
+  }, [environment, commitSha, commitMessage, pullRequestId, branch, repoOwner, repoSlug])
```

Comment:
All of this can be moved outside of the component and the `useEffect` can then be removed :)

Reply:
i think the useEffect is required to stop it re-running, as the component keeps re-rendering. 

I don't mind how we do it, these are the 2 requirements I think:

- console log only runs once
- self host (ie `!isPlatform`) should not be burdened by anything, or see any reference to this. even a stray useState set as `false` to hide something is little off-putting.

Reply:
@alaister bumpy bump

Reply:
I meant move it outside of the component completely – but this is actually fine – can merge!